concurrency/runner.go

```diff
@@ -0,0 +1,108 @@
+package concurrency
+
+import (
+	"context"
+	"sync"
+
+	tsdbErrors "github.com/prometheus/prometheus/tsdb/errors"
```

Comment:
Sadly we use Prometheus/Prometheus here. Is there a better error collector type object to use here? Here's the [Prometheus implementation of multierror](https://github.com/prometheus/prometheus/blob/main/tsdb/errors/errors.go#L27)

Reply:
Let's use `multierror.MultiError` from ... `dskit` :)

Reply:
Oh, Doh!